### PR TITLE
[Hotfix] #177 Fix registration status and program assignment

### PR DESCRIPTION
**Pull request by Satendra-SR**
#177

### Types of changes

- [x] Bug fix (a change which fixes an issue)
- [ ] New feature (a change which adds functionality)

### Short description of what this resolves:

While updating a registration, the program id and status are not getting updated correctly.

### Checklist:

- [ ] I have performed a self-review of my own code.
- [ ] The code follows the style guidelines of this project.
- [ ] The code changes are passing the CI checks
- [ ] I have documented my code wherever required.
- [ ] The changes requires a change to the documentation.
- [ ] I have updated the documentation based on the my changes.

### CODE REVIEW

File: api/services/registration_service.py [hunk 1]
```diff
@@ -85,15 +85,21 @@ def update_registration(self, registration, jsonData):
             registration (object): Takes registration as object
             jsonData (json): Takes request json for updating registration fields
         """
-        if not self.has_default_program_selection:
-            registration.program_id = self.selected_program_id
-            registration.status = models.Registration.RegistrationStatus.COMPLETE
-            registration.signup_date = datetime.now()
+        if not registration:
```
Comment (Satendra-SR) [inline api/services/registration_service.py:88]: Return if registration model object is not available.

File: api/services/registration_service.py [hunk 2]
```diff
@@ -85,15 +85,21 @@ def update_registration(self, registration, jsonData):
...
+            return
 
-        if registration:
-            self.user_id = (
-                self.create_user(jsonData) if self.selected_program_id else None
-            )
+        registration.program_id = self.selected_program_id
```
Comment (Satendra-SR) [inline api/services/registration_service.py:91]: Add program id whether it is selected or default

File: api/services/registration_service.py [hunk 3]
```diff
@@ -85,15 +85,21 @@ def update_registration(self, registration, jsonData):
...
+
+        self.user_id = self.create_user(jsonData) if self.selected_program_id else None
+        registration.signup_date = (
```
Comment (Satendra-SR) [inline api/services/registration_service.py:94]: Add signup_date only if program id is selected from prompt

File: api/services/registration_service.py [hunk 4]
```diff
@@ -85,15 +85,21 @@ def update_registration(self, registration, jsonData):
...
+            None if self.has_default_program_selection else datetime.now()
+        )
+
+        registration.status = (
+            models.Registration.RegistrationStatus.INCOMPLETE
+            if self.has_default_program_selection
+            else models.Registration.RegistrationStatus.COMPLETE
+        )
```
Comment (Satendra-SR) [inline api/services/registration_service.py:102]: Marking registration status as `incomplete` in case of default program id.

**Review by GauravGusain98 — DISMISSED** [review]
Looks good to me.

**Review by GauravGusain98 — APPROVED** [review]

**Pull request merged**